lib/request.js

```diff
@@ -220,25 +220,114 @@ Object.defineProperties(Request.prototype, {
   },
   hostname: {
     get () {
-      return this.host.split(':', 1)[0]
+      const host = this.host
+
+      /* Account for `requireHostHeader` set to `false` in Node.js */
+      if (!host) {
+        return ''
+      }
+
+      /* IPv6-specific behavior */
+
+      if (host.startsWith('[')) {
+        const closeBracketIndex = host.indexOf(']')
+
+        /* No closing bracket */
+        if (closeBracketIndex === -1) {
+          return ''
+        }
+
+        /* No hostname */
+        if (closeBracketIndex === 1) {
+          return ''
+        }
```

Comment:
How is this possible? Seems like such a request wouldn't even be routable, much less be processed by Fastify.

Reply:
I am completely new to this code base and therefore do not know exactly how the parsing takes place here. For example, I have in mind that the host should also be partially taken as a fallback from the :authority header. It is therefore possible that the user could send strange values here that are nevertheless routed. I therefore preferred to play it safe and made the checks as robust as possible. Please have a look at my tests and see if you approve the behavior.